lua/opencode/ui/output_window.lua

```diff
@@ -155,6 +157,36 @@ function M.setup_keymaps(windows)
 end
 
 function M.setup_autocmds(windows, group)
+  local guard_timer = nil
+
+  local function start_guard_timer()
+    if guard_timer then
+      return -- Timer already running
+    end
+    guard_timer = Timer.new({
```

Comment:
Is the poupose of the timer only to check the gard status ?

Whan can then guard change ? When files are added ?

If it's the case then having an event being sent whan the context files changes, would be better.

Then the topbar could listen to that event ?

```lua

-- In context.lua add files
state.event_manager:emit('custom.context.file.added')


--- in topbar
state.event_manager:subscribe('custom.context.file.added', M.render)
```

Having a timer check a user function every second could lead to bad performance if the function do blocking things.

The worst case would be an outdated topbar. but since the guard is checked again before sending the message it would not be that bad ?

What do you think ?

Reply:
I think we should start small and simple.  

You can subscribe to state.is_opencode_focused as well



If it becomes an issue that the toolbar is not in sync we can always address it later.